spec/formatting.md

```diff
@@ -18,3 +18,38 @@ the local variable takes precedence.
 
 It is an error for a local variable definition to
 refer to a local variable that's defined after it in the message.
+
+## Bidirectional Isolation
+
+Formatted messages may be bidirectional,
+i.e. contain a mix of left-to-right and right-to-left content.
+In all output formats,
+such parts should be isolated from each other so that
+the directionality of one part of the message does not affect
+the presentation of the rest of the message.
```

Comment:
I think this doesn't quite explain the issue. Perhaps:

> Formatted messages can be bidirectional, that is, containing a mixture of left-to-right and right-to-left text. When replacing placeholders with formatted string values, the [Unicode Bidirectional Algorithm](http://www.unicode.org/reports/tr9/) \[UAX9\] can produce unexpected or undesirable effects, such as "spillover". In such cases, placeholders _should_ (must?) be bidirectionally "isolated" so that the directionality of the text in the placeholder does not negatively affect the presentation of the overall string.

We might also link to some of the work Richard and I have been doing, e.g. https://www.w3.org/International/articles/lang-bidi-use-cases/#bidiCase7

Reply:
```suggestion
Formatted messages can be bidirectional,
containing a mixture of left-to-right and right-to-left text.
When replacing placeholders with formatted string values,
the [Unicode Bidirectional Algorithm](http://www.unicode.org/reports/tr9/) [UAX9]
can produce unexpected or undesirable effects, such as "spillover".
In such cases, placeholders should be bidirectionally "isolated"
so that the directionality of the text in the placeholder
does not negatively affect the presentation of the overall string.
```

I think it does need to be "should", because "must" would preclude an implementation from providing any isolation strategy that carries any risk of not isolating parts. And that feels like a bit of an overreach.

Btw, I'm using the leftmost "Add a suggestion" button of the comment editing toolbar here to enter a committable suggestion.